### ISSUE TITLE
Installing libutil after failing a cargo build results in missing headers

### ISSUE THREAD

**Issue by D1plo1d** [andrewshadura/bsd-pidfile-rs#1]
#### Observed Behaviour
On Ubuntu 20.04 installing `sudo apt install libutil-freebsd-dev` after trying to run `cargo build` resulted in a long error containing the following:

`ld.lld: error: undefined symbol: pidfile_open`

My guess is that the cargo build process was not failing early enough so it thought that `pidfile-rs` was successfully installed despite the missing header files (libutil was not installed on the first `cargo build`).

#### Expected Behaviour

It would be helpful if either this could fail earlier in the build process so that cargo doesn't assume a successful build of `pidfile-rs` when `libutil` is missing.

##### Work Arounds

If anyone else encounters this issue the solution is to install libutil-freebsd-dev and then run `cargo clean -p pidfile-rs`. After that your builds should succeed.

**Comment by andrewshadura** [comment]
I think it’s worth noting that at the moment the only implementation really supported is that in libbsd, as it provides a pkg-config file. To be fair, I’m a bit surprised something worked at all in your case — but re-checking the code it seems I never use the result of the pkg-config check :slightly_smiling_face:

**Comment by D1plo1d** [comment]
haha! Yeah, can confirm it looks like the pidfile checks are working as expected. Yay, unexpected functionality! :laughing:

**Comment by D1plo1d** [comment]
So it's my first time using either libbsd or libutil is there any major differences that I cause issue later or is it essentially a re-packaging of the same thing?

**Comment by andrewshadura** [comment]
As far as I understand, libbsd is a portable version of utility functions from *BSD systems, while `libutil-freebsd-dev` is a package providing the original library with all original semantics (including the header names and the lack of pkg-config).

**Comment by D1plo1d** [comment]
Sorry, not being a C++ developer that distinction is going a little over my head atm but it sounds like this is the officially supported library I want yeah?

https://packages.ubuntu.com/focal/libbsd-dev

**Comment by andrewshadura** [comment]
Yes, you want to install `libbsd-dev` on Debian or Ubuntu to get this to work.

**Comment by D1plo1d** [comment]
Awesome, I'll update my stuff. Thanks @andrewshadura !
